### DISCUSSION TITLE
Add annotations of type "GFF, BED and Narrowpeak" to assemblies using the cli,

### DISCUSSION THREAD

**Discussion by Alber-Tadrous** [GMOD/jbrowse-components#3265] [Q&A]
I want to add annotation to assemblies using the cli, is it possible?

**Comment by cmdcolin** [comment]
The "jbrowse add-track" command can add annotation tracks (gff, bed, bam, cram, bigwig, etc) similar to how "jbrowse add-assembly" can add the assembly (fasta, 2bit, etc) https://jbrowse.org/jb2/docs/tutorials/config_cli/

**Comment by Alber-Tadrous** [comment]
But "jbrowse add-track" doesn't have an option to specify the adapter type like the web page, so I always get this error "_Error: Track type is not supported_" when I try to add some of these file extensions throw cli command

**Reply by cmdcolin** [reply]
What file type are you trying to add? It is fairly picky about the file extension matching what it knows, which is something we could try to improve on

**Reply by Alber-Tadrous** [reply]
Last time I tried to add .bed track using cli I got this error, but when I added it from the web page it was added successfully

**Reply by cmdcolin** [reply]
the ability to add plaintext BED files with the CLI was added fairly recently (v2.1.3) xref https://jbrowse.org/jb2/blog/#213-2022-09-15

you can update your CLI by re-running e.g. npm install -g @jbrowse/cli

before we added this, we  expected people to do tabix indexing of BED.

also, I acknowledge it is a little odd also that it worked via web but not CLI, but the code for adding tracks in the CLI is a bit different from the web interface currently

**Reply by Alber-Tadrous** [reply]
".bed" files problem resolved after upgrading to the latest version, but I have another question, does jbrowse support ".narrowPeak" files

**Reply by cmdcolin** [reply]
if you make the narrowPeak file into a "bigBed" file, then jbrowse will be able to load it easily. bigBed is a binary version of BED that contains the column names in it, and is a good way for jbrowse to load the many BED-derivative file types. you can see instructions for creating a bigbed from a narrowpeak file here https://genome.ucsc.edu/goldenPath/help/bigNarrowPeak.html

basically, download bedToBigBed from here e.g. for linux http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedToBigBed

```
wget http://hgdownload.soe.ucsc.edu/admin/exe/linux.x86_64/bedToBigBed
chmod +x bedToBigBed
wget https://genome.ucsc.edu/goldenPath/help/examples/bigNarrowPeak.as
sort -k1,1 -k2,2n input.narrowpeak.bed > output.narrowpeak.bed
bedToBigBed -as=bigNarrowPeak.as -type=bed6+4 output.narrowpeak.bed chrom.sizes output.narrowpeak.bb
jbrowse add-track output.narrrowpeak.bb
```

alternatively, if you want to keep plaintext BED and do not want to use bigBed, you can rename the narrowPeak to .bed, and use jbrowse add-track on the .bed file, and possibly add a header line to the BED file to  specify the column names in the file (this will help jbrowse understand what each column in the bed means)

e.g.

```
#chrom chromStart chromEnd name score strand signalValue pValue qValue peak
```

**Reply by Alber-Tadrous** [reply]
Thank you so much, appreciate your help
